fem/fespace.cpp

```diff
@@ -561,6 +561,68 @@ void FiniteElementSpace::GetEssentialVDofs(const Array<int> &bdr_attr_is_ess,
    }
 }
 
+void FiniteElementSpace::GetEssentialVDofs(const Array<int> &bdr_attr_is_ess,
+                                           Array<int> &ess_vdofs,
+                                           const Array2D<int> &component) const
+{
+   Array<int> vdofs, dofs, ess_comp;
+
+   bool cmpID_rows = false;
+
+   ess_vdofs.SetSize(GetVSize());
+   ess_vdofs = 0;
+
+   ess_comp.SetSize(3);
+   ess_comp = 0;
+
+   for (int i = 0; i < GetNBE(); i++)
+   {
+      int id = GetBdrAttribute(i)-1;
+      if (bdr_attr_is_ess[id])
+      {
+         cmpID_rows = true;
+         //Checking to see if all the values in the row are greater than -1 if so then
+         //all components on a boundary are said to be marked.
+         for (int j = 0; j < component.NumCols(); ++j)
+         {
+            cmpID_rows = cmpID_rows && ( component(id, j) > -1);
+         }
+         if (cmpID_rows) // same as srw component id system
```

Comment:
what is srw?

Reply:
Sorry I missed that comment. I thought I'd gotten most of them. It's the initials of one the initial contributors to ExaConstit.